Kudu.Core/SiteExtensions/SiteExtensionStatus.cs

```diff
@@ -101,6 +100,26 @@ public bool IsTerminalStatus()
                 || string.Equals(Constants.SiteExtensionProvisioningStateCanceled, ProvisioningState, StringComparison.OrdinalIgnoreCase);
         }
 
+        private string SafeRead(string key)
+        {
+            try
+            {
+                return _jsonSettings.GetValue(key);
+            }
+            catch
+            {
+                // if setting file happen to be invalid, e.g w3wp.exe was kill while writting
+                // treat it as failed, and suggest user to re-install or un-install
+                _jsonSettings.SetValues(new KeyValuePair<string, JToken>[] {
```

Comment:
for write, if failed, i will clear all the content, and write the new key/value into json file.